src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalysisResultBuilder.cs

```diff
@@ -348,8 +348,9 @@ internal AnalysisResult ToAnalysisResult(ImmutableArray<DiagnosticAnalyzer> anal
                 foreach (var analyzer in analyzers)
                 {
                     var actionCounts = _analyzerActionCounts[analyzer];
+                    var suppressionActionCounts = analyzer is DiagnosticSuppressor ? 1 : 0;
```

Comment:
>suppressionActionCounts [](start = 24, length = 23)

Is it never greater than 1?

Reply:
Yes, suppressors cannot register multiple callbacks, just override the ReportSuppressions method.